server/controllers/__tests__/postQueryController.test.js

```diff
@@ -0,0 +1,123 @@
+jest.mock('../../services/Parser')
+
+
+const res = {
+    json: jest.fn()
+}
+
+const log = jest.spyOn(global.console, "log").mockImplementation(() => {})
+
+beforeEach(() => {
+    jest.resetAllMocks()
+})
+
+describe('postquery controller test suite', ()=> {
+    const getDateMock = jest.fn()
+    const getMealtimeMock = jest.fn()
+    const getStopwordMock = jest.fn()
+    
+    test('should get the values provided in the query', () => {
+        const req = {query: {query: 'abc, not lunch today!' }}
+        const newFixture = [
+            {
+              "matchedWord": "abc",
+              "type": "unknown",
+              "value": "abc"
+            },
+            {
+              "matchedWord": "not",
+              "type": "stopword",
+              "value": "not"
+            },
+            {
+              "matchedWord": "lunch",
+              "type": "mealtime",
+              "value": "midday"
+            },
+            {
+              "matchedWord": "today",
+              "type": "date",
+              "value": "05/17/2021"
+            }
+          ]
+        
+        parseIt.mockImplementation(() => {
+            return {
+                getStopword: getStopwordMock,
+                getDate: getDateMock,
+                getMealtime: getMealtimeMock
+            }
+        
+        })
+
+        
+        getDateMock
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([{ value: '05/17/2021', type: 'date' }])
+
+
+        getMealtimeMock
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([{ value: 'midday', type: 'mealtime' }])
+        .mockReturnValueOnce([])
+
+        getStopwordMock
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([{ value: 'not', type: 'stopword' }])
+        .mockReturnValueOnce([])
+        .mockReturnValueOnce([])
+        
+        
+        controller(req, res)
+ 
+        expect(getDateMock).toBeCalled()
+        expect(getMealtimeMock).toBeCalled()
+        expect(getStopwordMock).toBeCalled()
+        expect(res.json).toBeCalledWith({
+            "status": "success",
+            "data": newFixture
+        })
+    })
+    //     it('should handle thrown error', () => {
+    //     const req = {query: { }}
+    //     const testError = new Error('TEST ERROR')
+    //     parseIt.mockImplementation(() => {
+    //         throw testError
+    //         // return {
+    //         //     getDate: getDateMock.mockImplementation(() => {throw testError}),
+    //         //     getMealtime: getMealtimeMock.mockImplementation(() => {throw testError}),
+    //         //     getStopword: getStopwordMock.mockImplementation(() => {throw testError})
+    //         // }
+    //     })
+
+    //     //controller(req, res)
+        
+    //    expect(() => {controller(req, res)}).toThrow()
+    // })
```

Comment:
Left over?

Reply:
no I have 2 error tests one for console logging error and another for the thrown error.